stateth/stateth.go

```diff
@@ -0,0 +1,257 @@
+	//command = strings.Split(fmt.Sprintf("docker run --network %s --name=%s_grafana -p %d:3000 -d grafana/grafana:5.1.3", se.config.DockerPrefix, se.config.DockerPrefix, se.config.GrafanaPort), " ")
+	command = strings.Split(fmt.Sprintf("docker run --network %s --name=%s_grafana -e GF_AUTH_ANONYMOUS_ENABLED=true -p %d:3000 -d grafana/grafana:5.1.3", se.config.DockerPrefix, se.config.DockerPrefix, se.config.GrafanaPort), " ")
+	r, err = exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(string(r))
+		return err
+	}
+	return nil
+}
+
+func (se *Stateth) cleanupContainers() error {
+	log.Info("removing influxdb container")
+	command := strings.Split(fmt.Sprintf("docker rm -f %s_influxdb", se.config.DockerPrefix), " ")
+	r, err := exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Warn(string(r))
+	}
+
+	log.Info("removing grafana container")
+	command = strings.Split(fmt.Sprintf("docker rm -f %s_grafana", se.config.DockerPrefix), " ")
+	r, err = exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Warn(string(r))
+	}
+
+	log.Info("removing network")
+	command = strings.Split(fmt.Sprintf("docker network rm %s", se.config.DockerPrefix), " ")
+	r, err = exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Warn(string(r))
+	}
+
+	return nil
+}
+
+func (se *Stateth) importGrafanaDatasource() error {
+	log.Info("importing grafana datasource")
+
+	dataSource := &gapi.DataSource{
+		Name:      "metrics",
+		Type:      "influxdb",
+		URL:       fmt.Sprintf("http://%s_influxdb:%d", se.config.DockerPrefix, se.config.InfluxDBPort),
+		Access:    "proxy",
+		Database:  "metrics",
+		User:      influxDBAdminUser,
+		Password:  influxDBAdminPass,
+		IsDefault: true,
+		BasicAuth: false,
+	}
+
+	_, err := se.gclient.NewDataSource(dataSource)
+	if err != nil {
+		log.Warn(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (se *Stateth) importGrafanaDashboards() error {
+	log.Info("importing grafana dashboards")
+
+	files, err := ioutil.ReadDir(se.config.DashboardsFolder)
+	if err != nil {
+		log.Warn(err.Error())
+		return nil
+	}
+
+	for _, f := range files {
+		name := f.Name()
+		if strings.Contains(name, "json") {
+			log.Info("importing dashboard", "dashboard", name)
+
+			blob, err := ioutil.ReadFile(filepath.Join(se.config.DashboardsFolder, name))
+			if err != nil {
+				log.Warn(err.Error())
+				return nil
+			}
+
+			model := se.prepareDashboardModel(string(blob))
+
+			_, err = se.gclient.SaveDashboard(model, false)
+			if err != nil {
+				log.Warn(err.Error())
+				return nil
+			}
+		}
+	}
+
+	return nil
+}
+
+func (se *Stateth) prepareDashboardModel(configJSON string) map[string]interface{} {
+	configMap := map[string]interface{}{}
+	err := json.Unmarshal([]byte(configJSON), &configMap)
+	if err != nil {
+		panic("invalid JSON got into prepare func")
+	}
+
+	delete(configMap, "id")
```

Comment:
We might be able to get a sticky `id` of dashboards if we don't remove the `id` and `uid`, allowing us to link specific dashboards right from the Geth Dashboard, rather than having to go first to the Grafana front page. We should experiment with this.